From 4af92f653b3b3263a5374bc2cfe7bab5bfe01bc7 Mon Sep 17 00:00:00 2001
From: Nick Saulnier <nsaulnier@ti.com>
Date: Wed, 20 Oct 2021 16:15:17 -0500
Subject: [PATCH 3/4] ARM: dts: am335x: Add PRU system events for virtio

A PRU system event "vring" has been added to each of the PRU nodes in
the PRU-ICSS remote processor subsystem to enable the virtio/rpmsg
communication between MPU and that PRU core. The additions are done
in the base am33xx-l4.dtsi, and so are inherited by all the AM335x
boards. Do note that PRUSS is available only on all AM3356+ SoCs.

The PRU system events is the preferred approach over using OMAP
mailboxes, as it eliminates an external peripheral access from
the PRU-side, and keeps the interrupt generation internal to the
PRUSS. The difference from MPU would be minimal in using one
versus the other.

Mailboxes can still be used if desired, but currently there is no
support on firmware-side for the SoC to use mailboxes. Either approach
would require that an appropriate firmware image is loaded/booted on
the PRU.

Signed-off-by: Nick Saulnier <nsaulnier@ti.com>
Signed-off-by: Suman Anna <s-anna@ti.com>
(cherry picked from commit be1be9177fcc19bbd8669e6378c1364814cc9092)
Patch-mainline: Queued in TI repository
Git-repo: https://git.ti.com/git/ti-linux-kernel/ti-linux-kernel.git
Git-commit: be1be9177fcc19bbd8669e6378c1364814cc9092
---
 arch/arm/boot/dts/am33xx-l4.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/am33xx-l4.dtsi b/arch/arm/boot/dts/am33xx-l4.dtsi
index 7da42a5b959c..158acfcda747 100644
--- a/arch/arm/boot/dts/am33xx-l4.dtsi
+++ b/arch/arm/boot/dts/am33xx-l4.dtsi
@@ -915,6 +915,9 @@ pru0: pru@34000 {
 					      <0x22400 0x100>;
 					reg-names = "iram", "control", "debug";
 					firmware-name = "am335x-pru0-fw";
+					interrupt-parent = <&pruss_intc>;
+					interrupts = <16 2 2>;
+					interrupt-names = "vring";
 				};
 
 				pru1: pru@38000 {
@@ -924,6 +927,9 @@ pru1: pru@38000 {
 					      <0x24400 0x100>;
 					reg-names = "iram", "control", "debug";
 					firmware-name = "am335x-pru1-fw";
+					interrupt-parent = <&pruss_intc>;
+					interrupts = <18 3 3>;
+					interrupt-names = "vring";
 				};
 
 				pruss_mdio: mdio@32400 {
-- 
2.35.3

